.github/scripts/install-ubuntu.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# Handles installing dependencies for the build process.  If an
+# install-ubuntu-$OS_VERSION.sh script exists, install-ubuntu-22.04.sh
+# for example, then that script is executed to install dependencies
+# for that particular build instead of the install-ubuntu-generic.sh
+# script.
+
+# Go three directories up to get the codebase root path.
+[[ -z $CODEBASE_ROOT ]] && CODEBASE_ROOT="$(dirname "$(dirname "$(dirname "$(readlink -f "$0")")")")"
+
+# This gets the Ubuntu codename & version from the local OS, or allows
+# it to be passed in as an environment variable.
+[[ -z $OS_CODENAME ]] && OS_CODENAME="$(grep 'VERSION_CODENAME=' /etc/os-release | cut -f2 -d'=' | tr -d '"')"
+[[ -z $OS_VERSION ]] && OS_VERSION="$(grep 'VERSION_ID=' /etc/os-release | cut -f2 -d'=' | tr -d '"')"
+
+# Runs a specific install script for an OS version if it exists or runs
+# the generic install script.
+if [[ -e "${CODEBASE_ROOT}/.github/scripts/install-ubuntu-${OS_VERSION}.sh" ]]; then
+    echo "::group::Installing $OS_CODENAME ($OS_VERSION) build dependencies"
+    # shellcheck disable=SC1090
+    source "${CODEBASE_ROOT}/.github/scripts/install-ubuntu-${OS_VERSION}.sh"
+else
+    echo "::group::Installing generic build dependencies"
+    # shellcheck source=./install-ubuntu-generic.sh
+    source "${CODEBASE_ROOT}/.github/scripts/install-ubuntu-generic.sh"
+fi
+echo "::endgroup::"
```

Comment:
Does this have some special meaning? "::endgroup::" is an odd thing to echo, and there's no "::begingroup::" anywhere that I can see.

Might just be a debugging assist, but it looks like nothing else here.

Reply:
It's a special output that gets picked up by the GitHub runner.  It encapsulates all of the output between a
```bash
echo "::group::My Grouped Output"
# do some things.
echo "::endgroup::"
```

And it looks like this in the runner output:
![image](https://user-images.githubusercontent.com/22217699/210250824-f9b1f0f5-705b-4e4c-98a6-f5be553c43e0.png)

Reply:
Now I know.

I think I've learned more from this review than you have, really. :upside_down_face:

Reply:
I've only learned these things recently myself!  I picked up on it from looking at what others were doing in their workflows, and I was like "oooh, nice!  That's clever!" I'm definitely not the originator of the ideas. 😊